libc/watcom/syscall/crt0.c

```diff
@@ -10,15 +10,18 @@
 #include <unistd.h>
 #include <errno.h>
 
-/* Watcom extern code refs are sym_, extern data refs are _sym */
+/*
+ * create STACK segment which is required for applications
+ * size is setup by linker to default size 4096 bytes
+ * or to explicitly defined size by linker directive "OPTION STACK=..."
+ */
+#pragma data_seg("STACK","STACK")
 
-/* external references created by Watcom C compilation - unused */
-int _argc;              /* with declaration of main() */
-int _8087;              /* when floating point seen */
+/* return back to default data segment */
+#pragma data_seg()
 
-/* cstart_ is an external reference created by Watcom C, pulls in asm/segments.asm */
-#pragma aux cstart_ "_*" modify [ bx cx dx si di ]
-extern void cstart_(void);
+/* ELKS modified calling convention */
+#pragma aux elkscall __modify [ __bx __cx __dx __si __di ]
```

Comment:
The purpose of this pragma aux modify, as well as the one below for `exit` is to eliminate the need for saving registers that OWC otherwise emits. This is not an or the ELKS calling convention but a way to lessen needless code that OWC produces in certain special cases either for functions only called in crt0.s or OS exit, in an attempt to produce a small executable. Thus, I am not sure whether it is a good idea to call this `elkscall` to avoid confusion. Something like `noregsave` might be better.

I have some other questions about how pragma aux modify might be used to remove push bp/mov bp,sp/pop bp sequences in other functions, but we can talk about that later.

Reply:
I used first what I have in mind I will change it as you proposed.

Reply:
Ok, go ahead with `noregsave` pragma aux alias instead of `elkscall` for "reducing unneeded register saves" in crt0.c functions, that will be easier understood to programmers what is being done. Approved.

It is OK to get rid of kluge `cstart_` here, but we have to ensure that watcom/asm/segments.asm always gets pulled in somehow. Previously, it was pulled in because OW defined this symbol. It seems it was never needed to be defined in crt0.c however.